packages/core/src/browser/widget-manager.ts

```diff
@@ -106,8 +106,13 @@ export class WidgetManager {
     protected async doGetWidget<T extends Widget>(key: string): Promise<T | undefined> {
```

Comment:
I still think this should be done where I've already made the changes. There is no need to do this in `getOrCreateWidget` because we only cache the widget promise (put it in the map) after the `await`. See [here](https://github.com/theia-ide/theia/pull/2424/files#diff-5ed17f1536f41e76b9984a4a73302e32R135). If the promise is rejected we do not cache it anyway. What do you think?

Reply:
No, you're right.